### ISSUE TITLE
Display impacted files for modules

### ISSUE THREAD

**Issue by DamnClin** [jhipster/jhipster-lite#2343]
In the module page it would be great to display the files that will be impacted by the module. The idea to do this for free (for module maintainers) will be to: 

* Create a project in a temp folder using module dependencies and parents (we need hierarchy for that);
* Apply the module;
* Get the diff to put the in the WebService responses and display that

**Comment by pascalgrimaud** [comment]
adding a big bounty here, as it seems to be a lot of work to achieve this, and it will be a great feature

**Comment by antarus** [comment]
Before going so far, we can already know the modified files since we list them in the factory.
"All you have to do" is extract the files defined in the module's factory with a new method.

Perhaps by defining `appendFiles` statically rather than dynamically.
If we add a dependency, we add the pom.xml to the list of affected files, etc.
but it makes redo part of the module `module`

**Comment by DamnClin** [comment]
I don't think it is really far :D. We do need to have modules hierarchy migrated to modules but that's the main pain point here I think

**Comment by antarus** [comment]
let's say that we have to execute everything twice to find out which files have been modified when they are known in the factory.

but having a diff at the end would be a plus :D

**Comment by pascalgrimaud** [comment]
The workaround for this ticket is simply exec the module without commiting so we can easily use git
So not sure it deserves to work on this
What do you think @DamnClin ?
